rclcpp/src/rclcpp/executor.cpp

```diff
@@ -729,18 +729,35 @@ Executor::wait_for_work(std::chrono::nanoseconds timeout)
   // Clear any previous wait result
   this->wait_result_.reset();
 
+  // we need to make sure that callback groups don't get out of scope
+  // during the wait. As in jazzy, they are not covered by the DynamicStorage,
+  // we explicitly hold them here as a bugfix
```

Comment:
i would add `TODO` if we already have a plan to proper fix in the future not to miss, this does not block the PR.